src/models/krate.rs

```diff
@@ -539,36 +517,37 @@ impl Crate {
     }
 }
 
-pub struct OwnerInviteEmail {
-    /// The destination email address for this email.
-    recipient_email_address: String,
+/// Details of a newly created invite.
+#[derive(Debug)]
+pub enum NewOwnerInvite {
+    /// The invitee was a [`User`], and they must accept the invite through the
+    /// UI or via the provided invite token.
+    User(User, Secret<String>),
 
-    /// Email body variables.
-    user_name: String,
-    domain: String,
-    crate_name: String,
-    token: SecretString,
+    /// The invitee was a [`Team`], and they were immediately added as an owner.
+    Team(Team),
 }
 
-impl OwnerInviteEmail {
-    pub fn recipient_email_address(&self) -> &str {
-        &self.recipient_email_address
-    }
+/// Error results from a [`Crate::owner_add()`] model call.
+#[derive(Debug, Error)]
+pub enum OwnerAddError {
+    /// An opaque [`BoxedAppError`].
+    #[error("{0}")] // AppError does not impl Error
+    AppError(BoxedAppError),
+
+    /// The requested invitee already has a pending invite.
+    ///
+    /// Note: Teams are always immediately added, so they cannot have a pending
+    /// invite to cause this error.
+    #[error("user already has pending invite")]
+    AlreadyInvited(User),
 }
```

Comment:
Importantly for https://github.com/rust-lang/crates.io/issues/2639, the error response separates out the "user already has an invite" condition. Previously this was an opaque success (string message).